### PR TITLE
CT: Allow log to reject expired certificates if configured

### LINKED ISSUE #597: CT: Add option to reject expired certs
This is likely something many non-archival style logs will want to do since it considerably cuts down on the amount of spam entries that can be added to the tree which balloon storage requirements.

### PR DESCRIPTION
**Pull request by rolandshoemaker**
Should also probably add a test.

Fixes #597.

### CODE REVIEW

File: examples/ct/instance.go [hunk 1]
```diff
@@ -83,7 +84,7 @@ func LogConfigFromFile(filename string) ([]LogConfig, error) {
 
 // SetUpInstance sets up a log instance that uses the specified client to communicate
 // with the Trillian RPC back end.
-func (cfg LogConfig) SetUpInstance(client trillian.TrillianLogClient, deadline time.Duration) (*PathHandlers, error) {
+func (cfg LogConfig) SetUpInstance(client trillian.TrillianLogClient, deadline time.Duration, rejectExpired bool) (*PathHandlers, error) {
```
Comment (daviddrysdale) [inline examples/ct/instance.go]: Shouldn't this just use `cfg.RejectExpired` rather than adding a new parameter?

File: examples/ct/handlers.go [hunk 2]
```diff
@@ -174,19 +174,21 @@ type LogContext struct {
 		allRsps *expvar.Map // http.rc => expvar.Int  (as "http-all-rsps")
 		rsps    *expvar.Map // entrypoint => expvar.Map[http.rc => expvar.Int]  (as "http-rsps")
 	}
+	rejectExpired bool
 }
 
 // NewLogContext creates a new instance of LogContext.
-func NewLogContext(logID int64, prefix string, trustedRoots *PEMCertPool, rpcClient trillian.TrillianLogClient, signer *crypto.Signer, rpcDeadline time.Duration, timeSource util.TimeSource) *LogContext {
+func NewLogContext(logID int64, prefix string, trustedRoots *PEMCertPool, rpcClient trillian.TrillianLogClient, signer *crypto.Signer, rpcDeadline time.Duration, timeSource util.TimeSource, rejectExpired bool) *LogContext {
```
Comment (daviddrysdale) [inline examples/ct/handlers.go]: Might be a smidgeon clearer if we shuffled `rejectExpired` to be just after `trustedRoots`, so the two validation-related arguments are together.

File: examples/ct/handlers.go [hunk 3]
```diff
@@ -174,19 +174,21 @@ type LogContext struct {
 		allRsps *expvar.Map // http.rc => expvar.Int  (as "http-all-rsps")
 		rsps    *expvar.Map // entrypoint => expvar.Map[http.rc => expvar.Int]  (as "http-rsps")
 	}
+	rejectExpired bool
```
Comment (daviddrysdale) [inline examples/ct/handlers.go]: Shuffle up to just after `trustedRoots` and add a comment?

File: examples/ct/instance.go (hunk 1 above)
Comment (rolandshoemaker) [inline examples/ct/instance.go]: Whoops, yup, obviously not paying enough attention.

**Pull request merged**